polyglot-connector-java/src/main/java/com/flipkart/polyglot/jdbc/PolyglotJDBCUrl.java

```diff
@@ -0,0 +1,210 @@
+
+/*
+ * Copyright (c) 2016. the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.flipkart.polyglot.jdbc;
+
+
+import com.flipkart.polyglot.utils.Constants;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.sql.SQLException;
+import java.util.Properties;
+import java.util.StringTokenizer;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.phoenix.util.PhoenixRuntime;
+
+/**
+ * Created by naveen.nahata on 23/03/16.
+ */
+public class PolyglotJDBCUrl {
+
+    private final String txnStroeUrl;
+    private final String archivalStoreUrl;
+    private final String url;
+    private final Constants.DataStoreType dataStoreType;
+
+    /**
+     * Constructor to return PolyglotJDBC object for url and info properties
+     * @param url
+     * @param info
+     * @throws SQLException
+     */
+    public PolyglotJDBCUrl(String url, Properties info) throws SQLException {
+        info = getURLParamProperties(url, info);
+
+                /* URL pattern e.g. jdbc:polyglot://username:password@txnstore:ip1:port1,ip2:port2@archivalstore:ip3:port3/keyspace/catalog?
+        property1=value1.. */
+
+        final Pattern p = Pattern
+                .compile("^jdbc:(polyglot)://((\\w+)(:(\\w*))?@)?(txnstore:(\\S+)@)?(archivalstore:(\\S+))?/(\\w+)/(\\w+)(\\?(\\S+))?");
```

Comment:
I think url pattern is tightly coupled with code written below, it cannot be used anywhere else.